packages/ethereum-storage/src/ethereum-storage.ts

```diff
@@ -391,6 +391,53 @@ export default class EthereumStorage implements StorageTypes.IStorage {
     return contentDataIdAndMeta;
   }
 
+  /**
+   * Try to get some previous ignored data
+   *
+   * @param options timestamp boundaries for the data retrieval
+   * @returns Promise resolving stored data
+   */
+  public async getIgnoredData(): Promise<StorageTypes.IEntry[]> {
+    if (!this.isInitialized) {
+      throw new Error('Ethereum storage must be initialized');
+    }
+    this.logger.info('Getting some previous ignored dataIds', ['ethereum']);
+
+    const ethereumEntries: StorageTypes.IEthereumEntry[] = await this.ignoredDataIds.getDataIdsToRetry();
+
+    // If no hash was found on ethereum, we return an empty list
+    if (!ethereumEntries.length) {
+      this.logger.info('No new data found.', ['ethereum']);
+      return [];
+    }
+
+    this.logger.debug('Fetching data from IPFS and checking correctness', ['ipfs']);
+
+    const entries = await ethereumEntriesToIpfsContent(
+      ethereumEntries,
+      this.ipfsManager,
+      this.ignoredDataIds,
+      this.logger,
+      this.maxConcurrency,
+    );
+
+    const ids = entries.map(entry => entry.id) || [];
+    // Pin data asynchronously
+    // tslint:disable-next-line:no-floating-promises
+    this.pinDataToIPFS(ids);
```

Comment:
why asynchronously? and what happens if it fails?

Reply:
Because the pining can be done afterward, If it fails it will just be warning in the logger.

Reply:
then we should track the warnings in the logger I guess ;) there are ways to create metrics or alerts for that in GCP logs

Reply:
Yes, it would a very nice to have! I Jira it !